### PR TITLE
ES: preference framework has moved out of v7 folder.

### PR DESCRIPTION
**Pull request by MarijnS95**
@jzoran What comparison operator do you propose in makefiles (https://github.com/sonyxperiadev/transpower/pull/5#discussion_r209561139)? I cannot find anything that doesn't involve `shell`; is there something built into AOSP?

See
https://android.googlesource.com/platform/frameworks/support/+/e94ca6676f8f845741369d4967255f642a4cefbb

### CODE REVIEW

**Comment by jzoran** [comment]
Here's a snippet from build/make/config.mk Google uses to set BOARD_SYSTEMSDK_VERSIONS:

# Set BOARD_SYSTEMSDK_VERSIONS to the latest SystemSDK version starting from P-launching
# devices if unset.
ifndef BOARD_SYSTEMSDK_VERSIONS
  ifdef PRODUCT_SHIPPING_API_LEVEL
  ifneq ($(call math_gt_or_eq,$(PRODUCT_SHIPPING_API_LEVEL),28),)
    ifeq (REL,$(PLATFORM_VERSION_CODENAME))
      BOARD_SYSTEMSDK_VERSIONS := $(PLATFORM_SDK_VERSION)
    else
      BOARD_SYSTEMSDK_VERSIONS := $(PLATFORM_VERSION_CODENAME)
    endif
  endif
  endif
endif

I do something similar, but let me finish up the tests and we can talk more.

**Comment by jerpelea** [comment]
please update the patch

File: Android.mk [hunk 1]
```diff
@@ -7,10 +7,8 @@ LOCAL_SRC_FILES := $(call all-java-files-under, src)
 
 LOCAL_PACKAGE_NAME := ExtendedSettings
 LOCAL_CERTIFICATE := platform
+LOCAL_PRIVATE_PLATFORM_APIS := true
```
Comment (jzoran) [inline Android.mk]: This change should be a separate commit.

File: Android.mk [hunk 2]
```diff
@@ -22,11 +20,19 @@ LOCAL_STATIC_JAVA_LIBRARIES := \
...
+ifeq ($(call math_gt_or_eq, $(PLATFORM_SDK_VERSION), 28), true)
+    LOCAL_RESOURCE_DIR += \
+        frameworks/support/preference/res
+else
+    LOCAL_RESOURCE_DIR += \
+        frameworks/support/v14/preference/res \
+        frameworks/support/v7/preference/res
+endif
+
+
```
Comment (jzoran) [inline Android.mk]: Remove extra line.

**Comment by jzoran** [comment]
Otherwise ok.
